<!DOCTYPE html>

<html lang="en">

<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <meta name="mobile-web-app-capable" content="yes">
    <title>
        BIT 150: Project 2 - HackMD
    </title>
    <link rel="icon" type="image/png" href="https://hackmd.io/favicon.png">
    <link rel="apple-touch-icon" href="https://hackmd.io/apple-touch-icon.png">

    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/ionicons/2.0.1/css/ionicons.min.css" integrity="sha256-3iu9jgsy9TpTwXKb7bNQzqWekRX7pPK+2OLj3R922fo=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/octicons/3.5.0/octicons.min.css" integrity="sha256-QiWfLIsCT02Sdwkogf6YMiQlj4NE84MKkzEMkZnMGdg=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.5.1/themes/prism.min.css" integrity="sha256-vtR0hSWRc3Tb26iuN2oZHt3KRUomwTufNIf5/4oeCyg=" crossorigin="anonymous" />
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@hackmd/emojify.js@2.1.0/dist/css/basic/emojify.min.css" integrity="sha256-UOrvMOsSDSrW6szVLe8ZDZezBxh5IoIfgTwdNDgTjiU=" crossorigin="anonymous" />
    <style>
        @import url(https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,500,500i|Source+Code+Pro:300,400,500|Source+Sans+Pro:300,300i,400,400i,600,600i|Source+Serif+Pro&subset=latin-ext);.hljs{background:#fff;color:#333;display:block;overflow-x:auto;padding:.5em}.hljs-comment,.hljs-meta{color:#969896}.hljs-emphasis,.hljs-quote,.hljs-string,.hljs-strong,.hljs-template-variable,.hljs-variable{color:#df5000}.hljs-keyword,.hljs-selector-tag,.hljs-type{color:#a71d5d}.hljs-attribute,.hljs-bullet,.hljs-literal,.hljs-number,.hljs-symbol{color:#0086b3}.hljs-built_in,.hljs-builtin-name{color:#005cc5}.hljs-name,.hljs-section{color:#63a35c}.hljs-tag{color:#333}.hljs-attr,.hljs-selector-attr,.hljs-selector-class,.hljs-selector-id,.hljs-selector-pseudo,.hljs-title{color:#795da3}.hljs-addition{background-color:#eaffea;color:#55a532}.hljs-deletion{background-color:#ffecec;color:#bd2c00}.hljs-link{text-decoration:underline}.markdown-body{word-wrap:break-word;font-size:16px;line-height:1.5}.markdown-body:after,.markdown-body:before{content:"";display:table}.markdown-body:after{clear:both}.markdown-body>:first-child{margin-top:0!important}.markdown-body>:last-child{margin-bottom:0!important}.markdown-body a:not([href]){color:inherit;text-decoration:none}.markdown-body .absent{color:#c00}.markdown-body .anchor{float:left;line-height:1;margin-left:-20px;padding-right:4px}.markdown-body .anchor:focus{outline:none}.markdown-body blockquote,.markdown-body dl,.markdown-body ol,.markdown-body p,.markdown-body pre,.markdown-body table,.markdown-body ul{margin-bottom:16px;margin-top:0}.markdown-body hr{background-color:#e7e7e7;border:0;height:.25em;margin:24px 0;padding:0}.markdown-body blockquote{border-left:.25em solid #ddd;color:#777;font-size:16px;padding:0 1em}.markdown-body blockquote>:first-child{margin-top:0}.markdown-body blockquote>:last-child{margin-bottom:0}.markdown-body kbd,.popover kbd{background-color:#fcfcfc;border:1px solid;border-color:#ccc #ccc #bbb;border-radius:3px;box-shadow:inset 0 -1px 0 #bbb;color:#555;display:inline-block;font-size:11px;line-height:10px;padding:3px 5px;vertical-align:middle}.markdown-body .loweralpha{list-style-type:lower-alpha}.markdown-body h1,.markdown-body h2,.markdown-body h3,.markdown-body h4,.markdown-body h5,.markdown-body h6{font-weight:600;line-height:1.25;margin-bottom:16px;margin-top:24px}.markdown-body h1 .octicon-link,.markdown-body h2 .octicon-link,.markdown-body h3 .octicon-link,.markdown-body h4 .octicon-link,.markdown-body h5 .octicon-link,.markdown-body h6 .octicon-link{color:#000;vertical-align:middle;visibility:hidden}.markdown-body h1:hover .anchor,.markdown-body h2:hover .anchor,.markdown-body h3:hover .anchor,.markdown-body h4:hover .anchor,.markdown-body h5:hover .anchor,.markdown-body h6:hover .anchor{text-decoration:none}.markdown-body h1:hover .anchor .octicon-link,.markdown-body h2:hover .anchor .octicon-link,.markdown-body h3:hover .anchor .octicon-link,.markdown-body h4:hover .anchor .octicon-link,.markdown-body h5:hover .anchor .octicon-link,.markdown-body h6:hover .anchor .octicon-link{visibility:visible}.markdown-body h1 code,.markdown-body h1 tt,.markdown-body h2 code,.markdown-body h2 tt,.markdown-body h3 code,.markdown-body h3 tt,.markdown-body h4 code,.markdown-body h4 tt,.markdown-body h5 code,.markdown-body h5 tt,.markdown-body h6 code,.markdown-body h6 tt{font-size:inherit}.markdown-body h1{font-size:2em}.markdown-body h1,.markdown-body h2{border-bottom:1px solid #eee;padding-bottom:.3em}.markdown-body h2{font-size:1.5em}.markdown-body h3{font-size:1.25em}.markdown-body h4{font-size:1em}.markdown-body h5{font-size:.875em}.markdown-body h6{color:#777;font-size:.85em}.markdown-body ol,.markdown-body ul{padding-left:2em}.markdown-body ol.no-list,.markdown-body ul.no-list{list-style-type:none;padding:0}.markdown-body ol ol,.markdown-body ol ul,.markdown-body ul ol,.markdown-body ul ul{margin-bottom:0;margin-top:0}.markdown-body li>p{margin-top:16px}.markdown-body li+li{padding-top:.25em}.markdown-body dl{padding:0}.markdown-body dl dt{font-size:1em;font-style:italic;font-weight:700;margin-top:16px;padding:0}.markdown-body dl dd{margin-bottom:16px;padding:0 16px}.markdown-body table{display:block;overflow:auto;width:100%;word-break:normal;word-break:keep-all}.markdown-body table th{font-weight:700}.markdown-body table td,.markdown-body table th{border:1px solid #ddd;padding:6px 13px}.markdown-body table tr{background-color:#fff;border-top:1px solid #ccc}.markdown-body table tr:nth-child(2n){background-color:#f8f8f8}.markdown-body img{background-color:#fff;box-sizing:initial;max-width:100%}.markdown-body img[align=right]{padding-left:20px}.markdown-body img[align=left]{padding-right:20px}.markdown-body .emoji{background-color:initial;max-width:none;vertical-align:text-top}.markdown-body span.frame{display:block;overflow:hidden}.markdown-body span.frame>span{border:1px solid #ddd;display:block;float:left;margin:13px 0 0;overflow:hidden;padding:7px;width:auto}.markdown-body span.frame span img{display:block;float:left}.markdown-body span.frame span span{clear:both;color:#333;display:block;padding:5px 0 0}.markdown-body span.align-center{clear:both;display:block;overflow:hidden}.markdown-body span.align-center>span{display:block;margin:13px auto 0;overflow:hidden;text-align:center}.markdown-body span.align-center span img{margin:0 auto;text-align:center}.markdown-body span.align-right{clear:both;display:block;overflow:hidden}.markdown-body span.align-right>span{display:block;margin:13px 0 0;overflow:hidden;text-align:right}.markdown-body span.align-right span img{margin:0;text-align:right}.markdown-body span.float-left{display:block;float:left;margin-right:13px;overflow:hidden}.markdown-body span.float-left span{margin:13px 0 0}.markdown-body span.float-right{display:block;float:right;margin-left:13px;overflow:hidden}.markdown-body span.float-right>span{display:block;margin:13px auto 0;overflow:hidden;text-align:right}.markdown-body code,.markdown-body tt{background-color:#0000000a;border-radius:3px;font-size:85%;margin:0;padding:.2em 0}.markdown-body code:after,.markdown-body code:before,.markdown-body tt:after,.markdown-body tt:before{content:"\00a0";letter-spacing:-.2em}.markdown-body code br,.markdown-body tt br{display:none}.markdown-body del code{text-decoration:inherit}.markdown-body pre{word-wrap:normal}.markdown-body pre>code{background:#0000;border:0;font-size:100%;margin:0;padding:0;white-space:pre;word-break:normal}.markdown-body .highlight{margin-bottom:16px}.markdown-body .highlight pre{margin-bottom:0;word-break:normal}.markdown-body .highlight pre,.markdown-body pre{background-color:#f7f7f7;border-radius:3px;font-size:85%;line-height:1.45;overflow:auto;padding:16px}.markdown-body pre code,.markdown-body pre tt{word-wrap:normal;background-color:initial;border:0;display:inline;line-height:inherit;margin:0;max-width:auto;overflow:visible;padding:0}.markdown-body pre code:after,.markdown-body pre code:before,.markdown-body pre tt:after,.markdown-body pre tt:before{content:normal}.markdown-body .csv-data td,.markdown-body .csv-data th{font-size:12px;line-height:1;overflow:hidden;padding:5px;text-align:left;white-space:nowrap}.markdown-body .csv-data .blob-line-num{background:#fff;border:0;padding:10px 8px 9px;text-align:right}.markdown-body .csv-data tr{border-top:0}.markdown-body .csv-data th{background:#f8f8f8;border-top:0;font-weight:700}.news .alert .markdown-body blockquote{border:0;padding:0 0 0 40px}.activity-tab .news .alert .commits,.activity-tab .news .markdown-body blockquote{padding-left:0}.task-list-item{list-style-type:none}.task-list-item label{font-weight:400}.task-list-item.enabled label{cursor:pointer}.task-list-item+.task-list-item{margin-top:3px}.task-list-item-checkbox{cursor:default!important;float:left;margin:.31em 0 .2em -1.3em!important;vertical-align:middle}.markdown-body{max-width:758px;overflow:visible!important;padding-bottom:40px;padding-top:40px;position:relative}.markdown-body .emoji{vertical-align:top}.markdown-body pre{border:inherit!important}.markdown-body code{color:inherit!important}.markdown-body pre code .wrapper{display:-moz-inline-flex;display:-ms-inline-flex;display:-o-inline-flex;display:inline-flex}.markdown-body pre code .gutter{float:left;overflow:hidden;-webkit-user-select:none;user-select:none}.markdown-body pre code .gutter.linenumber{border-right:3px solid #6ce26c!important;box-sizing:initial;color:#afafaf!important;cursor:default;display:inline-block;min-width:20px;padding:0 8px 0 0;position:relative;text-align:right;z-index:4}.markdown-body pre code .gutter.linenumber>span:before{content:attr(data-linenumber)}.markdown-body pre code .code{float:left;margin:0 0 0 16px}.markdown-body .gist .line-numbers{border-bottom:none;border-left:none;border-top:none}.markdown-body .gist .line-data{border:none}.markdown-body .gist table{border-collapse:inherit!important;border-spacing:0}.markdown-body code[data-gist-id]{background:none;padding:0}.markdown-body code[data-gist-id]:after,.markdown-body code[data-gist-id]:before{content:""}.markdown-body code[data-gist-id] .blob-num{border:unset}.markdown-body code[data-gist-id] table{margin-bottom:unset;overflow:unset}.markdown-body code[data-gist-id] table tr{background:unset}.markdown-body[dir=rtl] pre{direction:ltr}.markdown-body[dir=rtl] code{direction:ltr;unicode-bidi:embed}.markdown-body .alert>p:last-child{margin-bottom:0}.markdown-body pre.abc,.markdown-body pre.flow-chart,.markdown-body pre.graphviz,.markdown-body pre.mermaid,.markdown-body pre.sequence-diagram,.markdown-body pre.vega{background-color:inherit;border-radius:0;overflow:visible;text-align:center;white-space:inherit}.markdown-body pre.abc>code,.markdown-body pre.flow-chart>code,.markdown-body pre.graphviz>code,.markdown-body pre.mermaid>code,.markdown-body pre.sequence-diagram>code,.markdown-body pre.vega>code{text-align:left}.markdown-body pre.abc>svg,.markdown-body pre.flow-chart>svg,.markdown-body pre.graphviz>svg,.markdown-body pre.mermaid>svg,.markdown-body pre.sequence-diagram>svg,.markdown-body pre.vega>svg{height:100%;max-width:100%}.markdown-body pre>code.wrap{word-wrap:break-word;white-space:pre-wrap;white-space:-moz-pre-wrap;white-space:-pre-wrap;white-space:-o-pre-wrap}.markdown-body .alert>p:last-child,.markdown-body .alert>ul:last-child{margin-bottom:0}.markdown-body summary{display:list-item}.markdown-body summary:focus{outline:none}.markdown-body details summary{cursor:pointer}.markdown-body details:not([open])>:not(summary){display:none}.markdown-body figure{margin:1em 40px}.markdown-body .mark,.markdown-body mark{background-color:#fff1a7}.vimeo,.youtube{background-color:#000;background-position:50%;background-repeat:no-repeat;background-size:contain;cursor:pointer;display:table;overflow:hidden;text-align:center}.vimeo,.youtube{position:relative;width:100%}.youtube{padding-bottom:56.25%}.vimeo img{object-fit:contain;width:100%;z-index:0}.youtube img{object-fit:cover;z-index:0}.vimeo iframe,.youtube iframe,.youtube img{height:100%;left:0;position:absolute;top:0;width:100%}.vimeo iframe,.youtube iframe{vertical-align:middle;z-index:1}.vimeo .icon,.youtube .icon{color:#fff;height:auto;left:50%;opacity:.3;position:absolute;top:50%;transform:translate(-50%,-50%);transition:opacity .2s;width:auto;z-index:0}.vimeo:hover .icon,.youtube:hover .icon{opacity:.6;transition:opacity .2s}.slideshare .inner,.speakerdeck .inner{position:relative;width:100%}.slideshare .inner iframe,.speakerdeck .inner iframe{bottom:0;height:100%;left:0;position:absolute;right:0;top:0;width:100%}.figma{display:table;padding-bottom:56.25%;position:relative;width:100%}.figma iframe{border:1px solid #eee;bottom:0;height:100%;left:0;position:absolute;right:0;top:0;width:100%}.markmap-container{height:300px}.markmap-container>svg{height:100%;width:100%}.MJX_Assistive_MathML{display:none}#MathJax_Message{z-index:1000!important}.ui-infobar{color:#777;margin:25px auto -25px;max-width:760px;position:relative;z-index:2}.toc .invisable-node{list-style-type:none}.ui-toc{bottom:20px;position:fixed;z-index:998}.ui-toc.both-mode{margin-left:8px}.ui-toc.both-mode .ui-toc-label{border-bottom-left-radius:0;border-top-left-radius:0;height:40px;padding:10px 4px}.ui-toc-label{background-color:#e6e6e6;border:none;color:#868686;transition:opacity .2s}.ui-toc .open .ui-toc-label{color:#fff;opacity:1;transition:opacity .2s}.ui-toc-label:focus{background-color:#ccc;color:#000;opacity:.3}.ui-toc-label:hover{background-color:#ccc;opacity:1;transition:opacity .2s}.ui-toc-dropdown{margin-bottom:20px;margin-top:20px;max-height:70vh;max-width:45vw;overflow:auto;padding-left:10px;padding-right:10px;text-align:inherit;width:25vw}.ui-toc-dropdown>.toc{max-height:calc(70vh - 100px);overflow:auto}.ui-toc-dropdown[dir=rtl] .nav{letter-spacing:.0029em;padding-right:0}.ui-toc-dropdown a{overflow:hidden;text-overflow:ellipsis;white-space:pre}.ui-toc-dropdown .nav>li>a{color:#767676;display:block;font-size:13px;font-weight:500;padding:4px 20px}.ui-toc-dropdown .nav>li:first-child:last-child>ul,.ui-toc-dropdown .toc.expand ul{display:block}.ui-toc-dropdown .nav>li>a:focus,.ui-toc-dropdown .nav>li>a:hover{background-color:initial;border-left:1px solid #000;color:#000;padding-left:19px;text-decoration:none}.ui-toc-dropdown[dir=rtl] .nav>li>a:focus,.ui-toc-dropdown[dir=rtl] .nav>li>a:hover{border-left:none;border-right:1px solid #000;padding-right:19px}.ui-toc-dropdown .nav>.active:focus>a,.ui-toc-dropdown .nav>.active:hover>a,.ui-toc-dropdown .nav>.active>a{background-color:initial;border-left:2px solid #000;color:#000;font-weight:700;padding-left:18px}.ui-toc-dropdown[dir=rtl] .nav>.active:focus>a,.ui-toc-dropdown[dir=rtl] .nav>.active:hover>a,.ui-toc-dropdown[dir=rtl] .nav>.active>a{border-left:none;border-right:2px solid #000;padding-right:18px}.ui-toc-dropdown .nav .nav{display:none;padding-bottom:10px}.ui-toc-dropdown .nav>.active>ul{display:block}.ui-toc-dropdown .nav .nav>li>a{font-size:12px;font-weight:400;padding-bottom:1px;padding-left:30px;padding-top:1px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>a{padding-right:30px}.ui-toc-dropdown .nav .nav>li>ul>li>a{font-size:12px;font-weight:400;padding-bottom:1px;padding-left:40px;padding-top:1px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>ul>li>a{padding-right:40px}.ui-toc-dropdown .nav .nav>li>a:focus,.ui-toc-dropdown .nav .nav>li>a:hover{padding-left:29px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>a:focus,.ui-toc-dropdown[dir=rtl] .nav .nav>li>a:hover{padding-right:29px}.ui-toc-dropdown .nav .nav>li>ul>li>a:focus,.ui-toc-dropdown .nav .nav>li>ul>li>a:hover{padding-left:39px}.ui-toc-dropdown[dir=rtl] .nav .nav>li>ul>li>a:focus,.ui-toc-dropdown[dir=rtl] .nav .nav>li>ul>li>a:hover{padding-right:39px}.ui-toc-dropdown .nav .nav>.active:focus>a,.ui-toc-dropdown .nav .nav>.active:hover>a,.ui-toc-dropdown .nav .nav>.active>a{font-weight:500;padding-left:28px}.ui-toc-dropdown[dir=rtl] .nav .nav>.active:focus>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active:hover>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active>a{padding-right:28px}.ui-toc-dropdown .nav .nav>.active>.nav>.active:focus>a,.ui-toc-dropdown .nav .nav>.active>.nav>.active:hover>a,.ui-toc-dropdown .nav .nav>.active>.nav>.active>a{font-weight:500;padding-left:38px}.ui-toc-dropdown[dir=rtl] .nav .nav>.active>.nav>.active:focus>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active>.nav>.active:hover>a,.ui-toc-dropdown[dir=rtl] .nav .nav>.active>.nav>.active>a{padding-right:38px}.markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang^=ja] .markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang=zh-tw] .markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang=zh-cn] .markdown-body{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html .markdown-body[lang^=ja]{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html .markdown-body[lang=zh-tw]{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html .markdown-body[lang=zh-cn]{font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica Neue,Helvetica,Roboto,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol}html[lang^=ja] .ui-toc-dropdown{font-family:Source Sans Pro,Helvetica,Arial,Meiryo UI,MS PGothic,ＭＳ Ｐゴシック,sans-serif}html[lang=zh-tw] .ui-toc-dropdown{font-family:Source Sans Pro,Helvetica,Arial,Microsoft JhengHei UI,微軟正黑UI,sans-serif}html[lang=zh-cn] .ui-toc-dropdown{font-family:Source Sans Pro,Helvetica,Arial,Microsoft YaHei UI,微软雅黑UI,sans-serif}html .ui-toc-dropdown[lang^=ja]{font-family:Source Sans Pro,Helvetica,Arial,Meiryo UI,MS PGothic,ＭＳ Ｐゴシック,sans-serif}html .ui-toc-dropdown[lang=zh-tw]{font-family:Source Sans Pro,Helvetica,Arial,Microsoft JhengHei UI,微軟正黑UI,sans-serif}html .ui-toc-dropdown[lang=zh-cn]{font-family:Source Sans Pro,Helvetica,Arial,Microsoft YaHei UI,微软雅黑UI,sans-serif}.ui-affix-toc{max-height:70vh;max-width:15vw;overflow:auto;position:fixed;top:0}.back-to-top,.expand-toggle,.go-to-bottom{color:#999;display:block;font-size:12px;font-weight:500;margin-left:10px;margin-top:10px;padding:4px 10px}.back-to-top:focus,.back-to-top:hover,.expand-toggle:focus,.expand-toggle:hover,.go-to-bottom:focus,.go-to-bottom:hover{color:#563d7c;text-decoration:none}.back-to-top,.go-to-bottom{margin-top:0}.ui-user-icon{background-position:50%;background-repeat:no-repeat;background-size:cover;border-radius:50%;display:block;height:20px;margin-bottom:2px;margin-right:5px;margin-top:2px;width:20px}.ui-user-icon.small{display:inline-block;height:18px;margin:0 0 .2em;vertical-align:middle;width:18px}.ui-infobar>small>span{line-height:22px}.ui-infobar>small .dropdown{display:inline-block}.ui-infobar>small .dropdown a:focus,.ui-infobar>small .dropdown a:hover{text-decoration:none}.ui-more-info{color:#888;cursor:pointer;vertical-align:middle}.ui-more-info .fa{font-size:16px}.ui-connectedGithub,.ui-published-note{color:#888}.ui-connectedGithub{line-height:23px;white-space:nowrap}.ui-connectedGithub a.file-path{color:#888;padding-left:22px;text-decoration:none}.ui-connectedGithub a.file-path:active,.ui-connectedGithub a.file-path:hover{color:#888;text-decoration:underline}.ui-connectedGithub .fa{font-size:20px}.ui-published-note .fa{font-size:20px;vertical-align:top}.unselectable{-webkit-user-select:none;-o-user-select:none;user-select:none}.selectable{-webkit-user-select:text;-o-user-select:text;user-select:text}.inline-spoiler-section{cursor:pointer}.inline-spoiler-section .spoiler-text{background-color:#333;border-radius:2px}.inline-spoiler-section .spoiler-text>*{opacity:0}.inline-spoiler-section .spoiler-img{filter:blur(10px)}.inline-spoiler-section.raw{background-color:#333;border-radius:2px}.inline-spoiler-section.raw>*{opacity:0}.inline-spoiler-section.unveil{cursor:auto}.inline-spoiler-section.unveil .spoiler-text{background-color:#3333331a}.inline-spoiler-section.unveil .spoiler-text>*{opacity:1}.inline-spoiler-section.unveil .spoiler-img{filter:none}@media print{blockquote,div,img,pre,table{page-break-inside:avoid!important}a[href]:after{font-size:12px!important}}.markdown-body.slides{color:#222;position:relative;z-index:1}.markdown-body.slides:before{background-color:currentColor;bottom:0;box-shadow:0 0 0 50vw;content:"";display:block;left:0;position:absolute;right:0;top:0;z-index:-1}.markdown-body.slides section[data-markdown]{background-color:#fff;margin-bottom:1.5em;position:relative;text-align:center}.markdown-body.slides section[data-markdown] code{text-align:left}.markdown-body.slides section[data-markdown]:before{content:"";display:block;padding-bottom:56.23%}.markdown-body.slides section[data-markdown]>div:first-child{left:1em;max-height:100%;overflow:hidden;position:absolute;right:1em;top:50%;transform:translateY(-50%)}.markdown-body.slides section[data-markdown]>ul{display:inline-block}.markdown-body.slides>section>section+section:after{border:3px solid #777;content:"";height:1.5em;position:absolute;right:1em;top:-1.5em}.site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,sans-serif}html[lang^=ja] .site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif}html[lang=zh-tw] .site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif}html[lang=zh-cn] .site-ui-font{font-family:Source Sans Pro,Helvetica,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif}body{font-smoothing:subpixel-antialiased!important;-webkit-font-smoothing:subpixel-antialiased!important;-moz-osx-font-smoothing:auto!important;-webkit-overflow-scrolling:touch;font-family:Source Sans Pro,Helvetica,Arial,sans-serif;letter-spacing:.025em}html[lang^=ja] body{font-family:Source Sans Pro,Helvetica,Arial,Hiragino Kaku Gothic Pro,ヒラギノ角ゴ Pro W3,Osaka,Meiryo,メイリオ,MS Gothic,ＭＳ ゴシック,sans-serif}html[lang=zh-tw] body{font-family:Source Sans Pro,Helvetica,Arial,PingFang TC,Microsoft JhengHei,微軟正黑,sans-serif}html[lang=zh-cn] body{font-family:Source Sans Pro,Helvetica,Arial,PingFang SC,Microsoft YaHei,微软雅黑,sans-serif}abbr[title]{border-bottom:none;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted}abbr[data-original-title],abbr[title]{cursor:help}body.modal-open{overflow-y:auto;padding-right:0!important}svg{text-shadow:none}
    </style>
    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
    	<script src="https://cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv.min.js" integrity="sha256-3Jy/GbSLrg0o9y5Z5n1uw0qxZECH7C6OQpVBgNFYa0g=" crossorigin="anonymous"></script>
    	<script src="https://cdnjs.cloudflare.com/ajax/libs/respond.js/1.4.2/respond.min.js" integrity="sha256-g6iAfvZp+nDQ2TdTR/VVKJf3bGro4ub5fvWSWVRi2NE=" crossorigin="anonymous"></script>
		<script src="https://cdnjs.cloudflare.com/ajax/libs/es5-shim/4.5.9/es5-shim.min.js" integrity="sha256-8E4Is26QH0bD52WoQpcB+R/tcWQtpzlCojrybUd7Mxo=" crossorigin="anonymous"></script>
    <![endif]-->
</head>

<body>
    <div id="doc" class="markdown-body container-fluid comment-enabled" data-hard-breaks="true"><h1 id="BIT-150-Project-2" data-id="BIT-150-Project-2"><a class="anchor hidden-xs" href="#BIT-150-Project-2" title="BIT-150-Project-2"><span class="octicon octicon-link"></span></a><span>BIT 150: Project 2</span></h1><h2 id="Introduction" data-id="Introduction"><a class="anchor hidden-xs" href="#Introduction" title="Introduction"><span class="octicon octicon-link"></span></a><span>Introduction</span></h2><p><span>The TP53 gene is a oncogene that is found on Chromosome 17, that provides instructions for making a protein known as Tumor Protein p53. The p53 protein binds to DNA and it makes another gene produce a protein(p21) that interacts with cell division simulating protein (cdk2). When this interaction occurs, the cell is not able to go through the cell divison process. But when the p53 protein is mutated, it cannot bind to DNA and the process to regulate cell division does not occur.   In order to have a properly funtioning TP53 gene, two functional copies of the TP53 gene need to be inherited. There is a condition called Li-Fraumeni syndrome where individuals inherit only one functional copy of the gene. Indivduals with this conditon are predisposed to cancer, and in their early adulthood, can start developing tumors in tissues thoughout their body.</span></p><p><em><span>Figure 1: The TP53 protein in teriary structure</span></em><br>
<img src="https://i.imgur.com/fPRUDW4.png" alt="" loading="lazy"></p><p><span>The biological feature we want to look at is the expression of the p53 protein throughout the body and compare it between cancerous and noncancerous tissues. By looking at the expression level of the protein in various tissues, we will be able to better understand the relationship between the TP53 gene and different cancers that are formed thoughout the body, which could help advance cancer treatments. Specifically, we will be looking at gene-disease association scores between TP53 and various types of cancers in addition to differential expression levels of TP53 in tissues throughout the body to better understand the relationship between TP53 and certain cancers.</span></p><p><strong><span>Our Research Question: “In what tissue(s) is cancer driven the most by TP53 expression?”</span></strong></p><h2 id="Methods" data-id="Methods"><a class="anchor hidden-xs" href="#Methods" title="Methods"><span class="octicon octicon-link"></span></a><span>Methods</span></h2><h3 id="Data-Acquisition" data-id="Data-Acquisition"><a class="anchor hidden-xs" href="#Data-Acquisition" title="Data-Acquisition"><span class="octicon octicon-link"></span></a><span>Data Acquisition</span></h3><h4 id="UCSC-Genome-Browser" data-id="UCSC-Genome-Browser"><a class="anchor hidden-xs" href="#UCSC-Genome-Browser" title="UCSC-Genome-Browser"><span class="octicon octicon-link"></span></a><span>UCSC Genome Browser</span></h4><p><span>We used the UCSC Genome Browser in order to find a graph that shows the amount of TP53 transcripts found in different parts of the body using the GTEx track. The individuals used for this study are noncancerous so it serves as a base for comparison with cancerous individuals.</span><br>
<span>Graph: </span><a href="https://www.genome.ucsc.edu/cgi-bin/hgc?hgsid=1502100755_GczbXh7k62Ou7KJX9Xtddg9Bjmhp&amp;db=hg38&amp;c=chr17&amp;l=7668420&amp;r=7687490&amp;o=7661778&amp;t=7677434&amp;g=gtexGeneV8&amp;i=TP53" target="_blank" rel="noopener"><span>https://www.genome.ucsc.edu/cgi-bin/hgc?hgsid=1502100755_GczbXh7k62Ou7KJX9Xtddg9Bjmhp&amp;db=hg38&amp;c=chr17&amp;l=7668420&amp;r=7687490&amp;o=7661778&amp;t=7677434&amp;g=gtexGeneV8&amp;i=TP53</span></a></p><h4 id="GENT2" data-id="GENT2"><a class="anchor hidden-xs" href="#GENT2" title="GENT2"><span class="octicon octicon-link"></span></a><span>GENT2</span></h4><p><img src="https://i.imgur.com/LPW27gG.png" alt="image alt" width="350" loading="lazy"><br>
<span>*source: </span><a href="http://gent2.appex.kr/gent2/" target="_blank" rel="noopener"><span>http://gent2.appex.kr/gent2/</span></a></p><p><span>We used the database GENT2 to get information about TP53 expression in normal and cancerous tissues throughout the body. The data was obtained by doing a gene profile search of TP53, which led to a page where the dataset could be downloaded. We then used this data to find the differential expression levels, helping us understand in which tissues TP53 expression levels differed more in cancerous individuals. We used the table provided by the website to calculate the differential expression levels between cancerous and healthy tissues.</span><br>
<span>Dataset: </span><a href="http://gent2.appex.kr/gent2/" target="_blank" rel="noopener"><span>http://gent2.appex.kr/gent2/</span></a></p><h4 id="DisGeNet" data-id="DisGeNet"><a class="anchor hidden-xs" href="#DisGeNet" title="DisGeNet"><span class="octicon octicon-link"></span></a><span>DisGeNet</span></h4><p><img src="https://i.imgur.com/cJnQqbD.png" alt="image alt" width="350" loading="lazy"><br>
<span>*source: </span><a href="https://www.disgenet.org/home/" target="_blank" rel="noopener"><span>https://www.disgenet.org/home/</span></a></p><p><span>We used the database DisGeNet to look at the association between diseases and genes. We were able to figure out the importance of TP53 in cancer prevention. We used the search bar to look up TP53. Results from the search provided us with information about the different gene disease assocation scores and the type of disease it was looking at. We used this information to identify the cancers most associated with the TP53 gene. We integrated this data with the information from the GENT2 database to make the comparison between cancerous and noncancerous tissues.</span><br>
<span>Dataset: </span><a href="https://www.disgenet.org/browser/1/1/0/7157/" target="_blank" rel="noopener"><span>https://www.disgenet.org/browser/1/1/0/7157/</span></a></p><h3 id="Data-Integration" data-id="Data-Integration"><a class="anchor hidden-xs" href="#Data-Integration" title="Data-Integration"><span class="octicon octicon-link"></span></a><span>Data Integration</span></h3><p><span>After downloading the relevant data from the DisGeNet and GENT2 databases, we used R to do our data integration and analysis. The first steps in this process were to load both datasets into R and select for the columns most relevant to our proposed research question.</span></p><pre><code># read in GENT2 expression data
differential_exp = read_table(file = "GENT2 Data.txt", col_names = TRUE)

# select for tissue and expression_level columns
differential_exp = select(differential_exp, Tissue, Expression_level)

# read in DisGeNet gene-disease association data
gene_disease_assoc = read_tsv(file= "DisGeNet Data.tsv")

# select for disease, disease_class, and score_gda columns
gene_disease_assoc = select(gene_disease_assoc, Disease, Disease_Class, Score_gda)
</code></pre><p><span>Once the data was read into R, we needed to clean up the datasets before integrating them. To accomplish this, we first split the Tissue column in the differential expression dataset from GENT2 into the Tissue and Type, which would clearly differentiate the tissue and whether it was normal or cancerous. The next step was to average the expression levels for each unique tissue type and put it into a new table. By doing this, we would only have one entry per tissue, making it easier to join with the DisGeNet table later on.</span></p><pre><code># split Tissue column into Tissue and Type (normal vs cancer) columns
differential_exp[c('Tissue', 'Type')] = str_split_fixed(differential_exp$Tissue,'-', 2)

# average expression level for each tissue for both normal and cancer types
avg_expression &lt;- aggregate(Expression_level ~ Tissue + Type, data = differential_exp, mean)
</code></pre><p><span>To make integrating the databases cleaner, we decided to first split the expression data into two tables, one for cancer and the other for normal tissue expression data. Then, we filtered for the tissues common between both tables and then joined the resulting two tables together by the Tissue column. This gave us a single table with each row being a unique Tissue; and, listed next to it, TP53’s average expression level in the normal and cancerous versions of that tissue. Furthermore, we replaced the underscores with spaces in the Tissue column of the average expressions data table. Also, we decided to keep only the Tissue, Normal Expression Level, and Cancer Expression Level columns as there is no longer a need to differentiate between Normal and Cancerous type tissues separately from the expression level columns. Finally, we added a new column in the avg_expression table of the differential expression levels between cancer and normal types of each tissue by taking the absolute value of the difference of expression levels between the cancer and normal tissue types.</span></p><pre><code># separate avg_expression into 2 tables- one for cancer tissues and one for normal tissues
# first split by Type column
split_avg_exp = split(avg_expression, avg_expression$Type)

# replace '_' with spaces for Tissue column
avg_expression$Tissue &lt;- str_replace_all(avg_expression$Tissue, "_", " ")

#assign cancer and normal tissue TP53 expression tables
cancer_avg_exp = as.data.frame(split_avg_exp[1])
colnames(cancer_avg_exp) &lt;- c("Cancer_Tissue", "Cancer_Type", "Cancer_Expression_Level")

normal_avg_exp = as.data.frame(split_avg_exp[2])
colnames(normal_avg_exp) &lt;- c("Tissue", "Normal_Type", "Normal_Expression_Level")

# normal_avg_exp has one extra tissue, filter this out prior to joining with cancer_avg_exp
normal_avg_exp = filter(normal_avg_exp, normal_avg_exp$Tissue %in% cancer_avg_exp$Cancer_Tissue)

# join normal and cancer average expression tables together
avg_expression = left_join(normal_avg_exp, cancer_avg_exp, by=c("Tissue" = "Cancer_Tissue"))

# remove the Normal_Type and Cancer_Type columns as no longer needed
avg_expression = select(avg_expression, Tissue, 'Normal_Expression_Level', 'Cancer_Expression_Level')

# add new column for differential expression (n)
avg_expression$Differential_Exp = abs(avg_expression$Cancer_Expression_Level - avg_expression$Normal_Expression_Level)
</code></pre><p><span>Then, we filtered the gene-disease association dataset from DisGeNet for disease names including the terms “malignant” or “carcinoma” as these were the specific types of cancerous diseases we are interested in investigating.</span></p><pre><code>gene_disease_assoc = filter(gene_disease_assoc, str_detect(Disease, "Malignant | Carcinoma"))
</code></pre><p><span>After the individual datasets were filtered for the relevant information, we had to prepare the datasets to be merged together. Since our project focuses on identifying the tissue(s) most related to TP53 expression, we decided to join the tables based on the tissue type. Doing so would allow us to observe the cancer association score from DisGeNet in addition to TP53 differential expression data from GENT2 across various tissues in the human body.</span></p><p><span>We decided to add a new column in the gene-disease association table that would denote the tissue type. To do this, we used a for loop that would extract the tissue type from various formats of disease names in the disease column, and then place that tissue name into the new Tissue column. At the end, we also capitalized the first letter of all the Tissue column values in order to ensure they would match the format of tissue type in the GENT2 expression dataset.</span></p><pre><code># add new column for Tissue in gene_disease_assoc data for those listed in format "neoplasm/tumor of ___" in Disease Column
for (i in 1:length(gene_disease_assoc$Disease)){
  if(grepl("neoplasm of | tumor of ", gene_disease_assoc$Disease[i])){
    gene_disease_assoc$Tissue[i] =  substr(gene_disease_assoc$Disease[i], gregexpr("of ", gene_disease_assoc$Disease[i])[[1]][1]+3, nchar(gene_disease_assoc$Disease[i]))
  }
  # for format " ___ Carcinoma"
  else if (grepl("Carcinoma", gene_disease_assoc$Disease[i])) {
    gene_disease_assoc$Tissue[i] =  substr(gene_disease_assoc$Disease[i], 0, gregexpr(" Carcinoma| Carcinoma ", gene_disease_assoc$Disease[i])[[1]][1]-1)
  }
  # for glioma, which corresponds to brain tissue
  else if (grepl("Glioma", gene_disease_assoc$Disease[i])) {
    gene_disease_assoc$Tissue[i] = "Brain"
  }
  # for format "Malignant ___ Neoplasm"
  else if (grepl("Neoplasm", gene_disease_assoc$Disease[i]) &amp; grepl("Malignant", gene_disease_assoc$Disease[i])) {
    gene_disease_assoc$Tissue[i] =  substr(gene_disease_assoc$Disease[i], gregexpr("Malignant", gene_disease_assoc$Disease[i])[[1]][1] + 10, gregexpr("Neoplasm", gene_disease_assoc$Disease[i])[[1]][1] -2)
  }
  else {
    gene_disease_assoc$Tissue[i] = ""
  }
}

# capitalize first letter of all Tissue values in gene_disease_assoc table
str_sub(gene_disease_assoc$Tissue, 0, 1) &lt;- str_to_upper(str_sub(gene_disease_assoc$Tissue, 0, 1))
</code></pre><p><span>Now that the individual datasets have been formatted, we then merged the Gene Disease Association and Average Expression tables into a single table.</span></p><pre><code># join avg_exp and gene_disease_assoc tables by Tissue 
# filter gene_disease_assoc for tissues in common with avg_expression
assoc_and_diff_exp &lt;- filter(gene_disease_assoc, gene_disease_assoc$Tissue %in% avg_expression$Tissue)

# join by Tissue column
assoc_and_diff_exp &lt;- left_join(assoc_and_diff_exp, avg_expression, by=c("Tissue"))
</code></pre><h3 id="Cytoscape" data-id="Cytoscape"><a class="anchor hidden-xs" href="#Cytoscape" title="Cytoscape"><span class="octicon octicon-link"></span></a><span>Cytoscape</span></h3><p><span>In order to visualize the association between the TP53 gene and expression in different body organs, we used Cytoscope to make a association map using the table produced by our data integration procedure. The map shows the level of importance that the TP53 gene has when it comes to gene-disease association in addition to the level at which TP53 is differentially expressed.</span></p><h2 id="Results" data-id="Results"><a class="anchor hidden-xs" href="#Results" title="Results"><span class="octicon octicon-link"></span></a><span>Results</span></h2><h3 id="Gene-Diesase-Association-and-Differential-Expression-Table" data-id="Gene-Diesase-Association-and-Differential-Expression-Table"><a class="anchor hidden-xs" href="#Gene-Diesase-Association-and-Differential-Expression-Table" title="Gene-Diesase-Association-and-Differential-Expression-Table"><span class="octicon octicon-link"></span></a><span>Gene-Diesase Association and Differential Expression Table</span></h3><p><span>The summary table including the information of gene-disease association (gda) and differential expression of various types of carcinomas and malignant neoplasms is shows below.</span></p><p><em><span>Table 1: Comparison of TP53 expression levels in cancerous VS noncancerous individuals.</span></em></p><table>
<thead>
<tr>
<th><span>Disease</span></th>
<th><span>Disease_Class</span></th>
<th><span>Score_gda</span></th>
<th><span>Tissue</span></th>
<th><span>Normal_Expression_Level</span></th>
<th><span>Cancer_Expression_Level</span></th>
<th><span>Differential_Exp</span></th>
</tr>
</thead>
<tbody>
<tr>
<td><span>Malignant neoplasm of breast</span></td>
<td><span>Neoplasms;    Skin and Connective   Tissue Diseases</span></td>
<td><span>1</span></td>
<td><span>Breast</span></td>
<td><span>8.209327</span></td>
<td><span>8.171658</span></td>
<td><span>0.037669</span></td>
</tr>
<tr>
<td><span>Breast Carcinoma</span></td>
<td><span>Neoplasms;    Skin and Connective   Tissue Diseases</span></td>
<td><span>0.9</span></td>
<td><span>Breast</span></td>
<td><span>8.209327</span></td>
<td><span>8.171658</span></td>
<td><span>0.037669</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of lung</span></td>
<td><span>Neoplasms;    Respiratory Tract   Diseases</span></td>
<td><span>0.8</span></td>
<td><span>Lung</span></td>
<td><span>7.69266</span></td>
<td><span>7.913639</span></td>
<td><span>0.220979</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of pancreas</span></td>
<td><span>Digestive System Diseases;      Neoplasms;    Endocrine System   Diseases</span></td>
<td><span>0.8</span></td>
<td><span>Pancreas</span></td>
<td><span>7.417076</span></td>
<td><span>7.799622</span></td>
<td><span>0.382546</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of esophagus</span></td>
<td><span>Digestive System Diseases;      Neoplasms</span></td>
<td><span>0.7</span></td>
<td><span>Esophagus</span></td>
<td><span>7.148531</span></td>
<td><span>7.044906</span></td>
<td><span>0.103625</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of ovary</span></td>
<td><span>Neoplasms;    Female Urogenital   Diseases and Pregnancy Complications;      Endocrine System Diseases</span></td>
<td><span>0.7</span></td>
<td><span>Ovary</span></td>
<td><span>7.929471</span></td>
<td><span>8.228522</span></td>
<td><span>0.299051</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of skin</span></td>
<td><span>Neoplasms;    Skin and Connective   Tissue Diseases</span></td>
<td><span>0.6</span></td>
<td><span>Skin</span></td>
<td><span>8.704935</span></td>
<td><span>9.318197</span></td>
<td><span>0.613262</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of liver</span></td>
<td><span>Digestive System Diseases;      Neoplasms</span></td>
<td><span>0.6</span></td>
<td><span>Liver</span></td>
<td><span>7.568792</span></td>
<td><span>7.686712</span></td>
<td><span>0.11792</span></td>
</tr>
<tr>
<td><span>Malignant Glioma</span></td>
<td><span>Neoplasms</span></td>
<td><span>0.6</span></td>
<td><span>Brain</span></td>
<td><span>6.548938</span></td>
<td><span>7.837003</span></td>
<td><span>1.288065</span></td>
</tr>
<tr>
<td><span>Malignant tumor of colon</span></td>
<td><span>Digestive System Diseases;      Neoplasms</span></td>
<td><span>0.5</span></td>
<td><span>Colon</span></td>
<td><span>8.297298</span></td>
<td><span>8.675002</span></td>
<td><span>0.377704</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of prostate</span></td>
<td><span>Neoplasms;    Male Urogenital   Diseases</span></td>
<td><span>0.5</span></td>
<td><span>Prostate</span></td>
<td><span>8.173235</span></td>
<td><span>9.245797</span></td>
<td><span>1.072562</span></td>
</tr>
<tr>
<td><span>Malignant tumor of cervix</span></td>
<td><span>Neoplasms;    Female Urogenital   Diseases and Pregnancy Complications</span></td>
<td><span>0.4</span></td>
<td><span>Cervix</span></td>
<td><span>8.758068</span></td>
<td><span>8.716564</span></td>
<td><span>0.041504</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of stomach</span></td>
<td><span>Digestive System Diseases;      Neoplasms</span></td>
<td><span>0.4</span></td>
<td><span>Stomach</span></td>
<td><span>7.405491</span></td>
<td><span>7.550245</span></td>
<td><span>0.144754</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of gallbladder</span></td>
<td><span>Digestive System Diseases;      Neoplasms</span></td>
<td><span>0.4</span></td>
<td><span>Gallbladder</span></td>
<td><span>8.735121</span></td>
<td><span>8.102161</span></td>
<td><span>0.63296</span></td>
</tr>
<tr>
<td><span>Malignant Head and Neck Neoplasm</span></td>
<td><span>Neoplasms</span></td>
<td><span>0.4</span></td>
<td><span>Head and Neck</span></td>
<td><span>7.967162</span></td>
<td><span>9.066948</span></td>
<td><span>1.099786</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of kidney</span></td>
<td><span>Neoplasms;    Female Urogenital   Diseases and Pregnancy Complications;      Male Urogenital Diseases</span></td>
<td><span>0.4</span></td>
<td><span>Kidney</span></td>
<td><span>7.401444</span></td>
<td><span>8.045522</span></td>
<td><span>0.644079</span></td>
</tr>
<tr>
<td><span>Head and Neck Carcinoma</span></td>
<td><span>Neoplasms</span></td>
<td><span>0.4</span></td>
<td><span>Head and Neck</span></td>
<td><span>7.967162</span></td>
<td><span>9.066948</span></td>
<td><span>1.099786</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of endometrium</span></td>
<td><span>Neoplasms;    Female Urogenital   Diseases and Pregnancy Complications</span></td>
<td><span>0.3</span></td>
<td><span>Endometrium</span></td>
<td><span>9.411807</span></td>
<td><span>8.75622</span></td>
<td><span>0.655587</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of tongue</span></td>
<td><span>Neoplasms;    Stomatognathic   Diseases</span></td>
<td><span>0.3</span></td>
<td><span>Tongue</span></td>
<td><span>8.079067</span></td>
<td><span>7.497676</span></td>
<td><span>0.581391</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of brain</span></td>
<td><span>Neoplasms;    Nervous System   Diseases</span></td>
<td><span>0.24</span></td>
<td><span>Brain</span></td>
<td><span>6.548938</span></td>
<td><span>7.837003</span></td>
<td><span>1.288065</span></td>
</tr>
<tr>
<td><span>Gallbladder Carcinoma</span></td>
<td><span>Digestive System Diseases;      Neoplasms</span></td>
<td><span>0.2</span></td>
<td><span>Gallbladder</span></td>
<td><span>8.735121</span></td>
<td><span>8.102161</span></td>
<td><span>0.63296</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of thyroid</span></td>
<td><span>Neoplasms;    Endocrine System   Diseases</span></td>
<td><span>0.1</span></td>
<td><span>Thyroid</span></td>
<td><span>7.962619</span></td>
<td><span>8.245334</span></td>
<td><span>0.282715</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of testis</span></td>
<td><span>Neoplasms;    Male Urogenital   Diseases;    Endocrine System Diseases</span></td>
<td><span>0.1</span></td>
<td><span>Testis</span></td>
<td><span>7.840079</span></td>
<td><span>8.072986</span></td>
<td><span>0.232907</span></td>
</tr>
<tr>
<td><span>Malignant neoplasm of vulva</span></td>
<td><span>Neoplasms;    Female Urogenital   Diseases and Pregnancy Complications</span></td>
<td><span>0.1</span></td>
<td><span>Vulva</span></td>
<td><span>8.803542</span></td>
<td><span>8.438418</span></td>
<td><span>0.365124</span></td>
</tr>
<tr>
<td><span>Tongue Carcinoma</span></td>
<td><span>Neoplasms;    Stomatognathic   Diseases</span></td>
<td><span>0.1</span></td>
<td><span>Tongue</span></td>
<td><span>8.079067</span></td>
<td><span>7.497676</span></td>
<td><span>0.581391</span></td>
</tr>
<tr>
<td><span>Colon Carcinoma</span></td>
<td><span>Digestive System Diseases;      Neoplasms</span></td>
<td><span>0.1</span></td>
<td><span>Colon</span></td>
<td><span>8.297298</span></td>
<td><span>8.675002</span></td>
<td><span>0.377704</span></td>
</tr>
<tr>
<td><span>Stomach Carcinoma</span></td>
<td><span>Digestive System Diseases;      Neoplasms</span></td>
<td><span>0.1</span></td>
<td><span>Stomach</span></td>
<td><span>7.405491</span></td>
<td><span>7.550245</span></td>
<td><span>0.144754</span></td>
</tr>
<tr>
<td><span>Primary Malignant Liver Neoplasm</span></td>
<td><span>Digestive System Diseases;      Neoplasms</span></td>
<td><span>0.04</span></td>
<td><span>Liver</span></td>
<td><span>7.568792</span></td>
<td><span>7.686712</span></td>
<td><span>0.11792</span></td>
</tr>
<tr>
<td><span>Vagina Carcinoma</span></td>
<td><span>Neoplasms;    Female Urogenital   Diseases and Pregnancy Complications</span></td>
<td><span>0.04</span></td>
<td><span>Vagina</span></td>
<td><span>8.461969</span></td>
<td><span>8.318347</span></td>
<td><span>0.143622</span></td>
</tr>
<tr>
<td><span>Malignant Bone Neoplasm</span></td>
<td><span>Neoplasms;    Musculoskeletal   Diseases</span></td>
<td><span>0.02</span></td>
<td><span>Bone</span></td>
<td><span>7.78284</span></td>
<td><span>7.367676</span></td>
<td><span>0.415164</span></td>
</tr>
<tr>
<td><span>Malignant epithelial tumor of ovary</span></td>
<td><span>Neoplasms;    Female Urogenital   Diseases and Pregnancy Complications;      Endocrine System Diseases</span></td>
<td><span>0.01</span></td>
<td><span>Ovary</span></td>
<td><span>7.929471</span></td>
<td><span>8.228522</span></td>
<td><span>0.299051</span></td>
</tr>
<tr>
<td><span>Colon Carcinoma Metastatic in the Liver</span></td>
<td><span>NA</span></td>
<td><span>0.01</span></td>
<td><span>Colon</span></td>
<td><span>8.297298</span></td>
<td><span>8.675002</span></td>
<td><span>0.377704</span></td>
</tr>
<tr>
<td><span>Childhood Malignant Liver Neoplasm</span></td>
<td><span>Digestive System Diseases;      Neoplasms</span></td>
<td><span>0.01</span></td>
<td><span>Liver</span></td>
<td><span>7.568792</span></td>
<td><span>7.686712</span></td>
<td><span>0.11792</span></td>
</tr>
</tbody>
</table><h3 id="Cytoscape-Visualization" data-id="Cytoscape-Visualization"><a class="anchor hidden-xs" href="#Cytoscape-Visualization" title="Cytoscape-Visualization"><span class="octicon octicon-link"></span></a><span>Cytoscape Visualization</span></h3><p><em><span>Figure 2: Cytoscope network showing TP53 gene expression in different tissues</span></em><br>
<img src="https://i.imgur.com/wxsvb5l.png" alt="" loading="lazy"><br>
<span>The node colors reflect the strength of association between TP53 and the disease name nodes.</span><br>
<span>The five colors of the nodes we used are:</span></p><p><span>Red = highest association</span><br>
<span>Red-Orange = high association</span><br>
<span>Orange = medium association</span><br>
<span>Orange-Yellow = low associaton</span><br>
<span>Yellow = lowest association</span></p><p><span>The edge colors depict the level at which TP53 is differentially expressed between the cancerous and normal versions of the tissue in the disease name nodes.</span><br>
<span>The colors we used for the edges are:</span></p><p><span>Dark Purple = highest difference in expression</span><br>
<span>Purplish-Blue = medium difference in expression</span><br>
<span>Blue = low difference in expression</span></p><h3 id="UCSC-Genome-Browser-Graph" data-id="UCSC-Genome-Browser-Graph"><a class="anchor hidden-xs" href="#UCSC-Genome-Browser-Graph" title="UCSC-Genome-Browser-Graph"><span class="octicon octicon-link"></span></a><span>UCSC Genome Browser Graph</span></h3><p><em><span>Figure 3: UCSC Genome Browser results of TP53 expression in noncancerous individuals</span></em><br>
<img src="https://i.imgur.com/Zw42Emp.jpg" alt="" loading="lazy"><br>
<span>*source: </span><a href="https://www.genome.ucsc.edu/cgi-bin/hgc?hgsid=1502100755_GczbXh7k62Ou7KJX9Xtddg9Bjmhp&amp;db=hg38&amp;c=chr17&amp;l=7668420&amp;r=7687490&amp;o=7661778&amp;t=7677434&amp;g=gtexGeneV8&amp;i=TP53" target="_blank" rel="noopener"><span>https://www.genome.ucsc.edu/cgi-bin/hgc?hgsid=1502100755_GczbXh7k62Ou7KJX9Xtddg9Bjmhp&amp;db=hg38&amp;c=chr17&amp;l=7668420&amp;r=7687490&amp;o=7661778&amp;t=7677434&amp;g=gtexGeneV8&amp;i=TP53</span></a></p><h3 id="GENT2-Graph-of-Differential-Expression" data-id="GENT2-Graph-of-Differential-Expression"><a class="anchor hidden-xs" href="#GENT2-Graph-of-Differential-Expression" title="GENT2-Graph-of-Differential-Expression"><span class="octicon octicon-link"></span></a><span>GENT2 Graph of Differential Expression</span></h3><p><em><span>Figure 4: Expression of TP53 in body organs of cancerous indivudals</span></em><br>
<img src="https://i.imgur.com/QGDesCf.png" alt="" loading="lazy"><br>
<span>*source: </span><a href="http://gent2.appex.kr/gent2/" target="_blank" rel="noopener"><span>http://gent2.appex.kr/gent2/</span></a></p><h2 id="Discussion" data-id="Discussion"><a class="anchor hidden-xs" href="#Discussion" title="Discussion"><span class="octicon octicon-link"></span></a><span>Discussion</span></h2><p><span>The combined Gene-Disease Association and Differential Expression table shows us the association scores between TP53 and the diseases as well as the difference in TP53 expression in noncancerous versus cancerous versions of those tissues. With this table, we can compare the gene-disease association (gda) scores with the differential expression and infer which disesase is most driven by TP53 as such diseases would have a high gda score and a large differential expression value in the associated tissue.</span></p><p><span>When looking at the genome of a noncancerous human with the TP53 gene, we see that the gene can be found in various parts of the body. When we look closely at the graph that we get from UCSC Genome Browser, we see that the trascripts per million for brain  cancers is really low compared to other cancers that are portrayed on the graph above. This can be due to the fact that the brain only uses TP53 when needed. It tries to maintain the cell division regaulrily but when it starts to get out of control, TP53 can come in and suppress the cell division. This is proven when we look at </span><strong><span>Table 1</span></strong><span> that shows expression level. It is low in </span><strong><span>Figure 3</span></strong><span> because the indiviudals tested were noncancerous whereras when looking at results in </span><strong><span>Table 1</span></strong><span>, the expression levels of TP53 is increased tremendously.</span></p><p><span>When looking at </span><strong><span>Table 1</span></strong><span> we can see an apparent difference in the expression levels for cancerous individuals compared to noncancerous individuals. After analysing the table, we concluded that the disease type that is most prevalent with mutations in the TP53 gene is Malignant gliomas. Malignant gliomas occur when there is an exponenital increase in cell divison in the brain. When we look at the gda value for malignant Gliomas in </span><strong><span>Table 1</span></strong><span> we see that it is 0.6, out of a maximum of 1.0, which means that there is a fairly strong association between the TP53 gene and this type of  cancer. Looking at the differential expression level showed us that there is a large difference in the expression of the gene between normal and healthy brain tissue. Based on </span><strong><span>Table 1</span></strong><span> there is a 1.28 difference in expression of the TP53 gene in indivudals with cancer compared to inviduals without this type of cancer. This confirms our beliefs that the malignant glioma is the cancer most associated to the TP53 gene as it has a fairly strong gene-disease association score and is the most differentially expressed in brain tissue.</span></p><p><span>This means that even though the TP53 gene is expressed in cells all over the body, it is expressed in different levels depending on the type of cancer the individuals has. The reason why the expression level in TP53 gene in an individual with Malignant Glioma is super high can be due to the fact that once these gliomas form, they can destroy the neurons in the brain that are responsible for bodily functions. The body starts producing such a high amount of p53 protein so that the tumor can be suppressed in the most important part of the body. Further analysis could be done to examine the levels of TP53 expression in various stages of malignant gliomas as this could provide more insight into the mechanisms through which TP53 expression increases to attempt to suppress the tumor.</span></p><h2 id="Conclusion" data-id="Conclusion"><a class="anchor hidden-xs" href="#Conclusion" title="Conclusion"><span class="octicon octicon-link"></span></a><span>Conclusion</span></h2><p><span>Through using diffent data types and genome networks, we found that the TP53 gene is most associated with individuals with Malignant Gliomas, which affects brain tissue. We were able to see a significatly higher level of expression in these indivudals due to the importance of the organ in daily bodily function. By comparing TP53 expression in noncancerous indiviudals compared to cancerous indiviudals we saw that in the brain, the overall amount of transctips for TP53 is relatively low. In malignant gliomas specifically, TP53 expression increases much more than in other cancers, suggesting it as a strong indicator of such cancers, and potentially others affecting the brain. In conclusion, when an indiviudal gets cancer, the amount of TP53 gene that is expressed increases on a need basis and is low when the individual is healthy. Further analysis on TP53 expression as cancerous tumors progress through stages could also help in providing a clearer picture into the details of changing TP53 expression in cancerous tissues.</span></p><h2 id="References" data-id="References"><a class="anchor hidden-xs" href="#References" title="References"><span class="octicon octicon-link"></span></a><span>References</span></h2><p><a href="https://medlineplus.gov/genetics/gene/tp53/" target="_blank" rel="noopener"><span>https://medlineplus.gov/genetics/gene/tp53/</span></a></p><p><a href="https://www.ncbi.nlm.nih.gov/books/NBK22268/" target="_blank" rel="noopener"><span>https://www.ncbi.nlm.nih.gov/books/NBK22268/</span></a></p><p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5625482/" target="_blank" rel="noopener"><span>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5625482/</span></a></p><p><a href="http://gent2.appex.kr/gent2/" target="_blank" rel="noopener"><span>http://gent2.appex.kr/gent2/</span></a></p><p><a href="https://www.disgenet.org/browser/1/1/0/7157/0/25/2494/_a/_b./-ei/" target="_blank" rel="noopener"><span>https://www.disgenet.org/browser/1/1/0/7157/0/25/2494/_a/_b./-ei/</span></a></p><p><a href="https://pubmed.ncbi.nlm.nih.gov/9804370/" target="_blank" rel="noopener"><span>https://pubmed.ncbi.nlm.nih.gov/9804370/</span></a></p><p><a href="https://www.frontiersin.org/articles/10.3389/fcell.2021.649891/full#:~:text=As%20with%20most%20tissues%20in,injury%20(Modo%2C%202019)" target="_blank" rel="noopener"><span>https://www.frontiersin.org/articles/10.3389/fcell.2021.649891/full#:~:text=As with most tissues in,injury (Modo%2C 2019)</span></a><span>.</span></p><p><a href="https://www.futuremedicine.com/doi/10.2217/bmm-2020-0331" target="_blank" rel="noopener"><span>https://www.futuremedicine.com/doi/10.2217/bmm-2020-0331</span></a></p></div>
    <div class="ui-toc dropup unselectable hidden-print" style="display:none;">
        <div class="pull-right dropdown">
            <a id="tocLabel" class="ui-toc-label btn btn-default" data-toggle="dropdown" href="#" role="button" aria-haspopup="true" aria-expanded="false" title="Table of content">
                <i class="fa fa-bars"></i>
            </a>
            <ul id="ui-toc" class="ui-toc-dropdown dropdown-menu" aria-labelledby="tocLabel">
                <div class="toc"><ul class="nav">
<li class=""><a href="#BIT-150-Project-2" title="BIT 150: Project 2">BIT 150: Project 2</a><ul class="nav">
<li class=""><a href="#Introduction" title="Introduction">Introduction</a></li>
<li class=""><a href="#Methods" title="Methods">Methods</a><ul class="nav">
<li class=""><a href="#Data-Acquisition" title="Data Acquisition">Data Acquisition</a></li>
<li class=""><a href="#Data-Integration" title="Data Integration">Data Integration</a></li>
<li class=""><a href="#Cytoscape" title="Cytoscape">Cytoscape</a></li>
</ul>
</li>
<li class=""><a href="#Results" title="Results">Results</a><ul class="nav">
<li class=""><a href="#Gene-Diesase-Association-and-Differential-Expression-Table" title="Gene-Diesase Association and Differential Expression Table">Gene-Diesase Association and Differential Expression Table</a></li>
<li class=""><a href="#Cytoscape-Visualization" title="Cytoscape Visualization">Cytoscape Visualization</a></li>
<li><a href="#UCSC-Genome-Browser-Graph" title="UCSC Genome Browser Graph">UCSC Genome Browser Graph</a></li>
<li><a href="#GENT2-Graph-of-Differential-Expression" title="GENT2 Graph of Differential Expression">GENT2 Graph of Differential Expression</a></li>
</ul>
</li>
<li><a href="#Discussion" title="Discussion">Discussion</a></li>
<li><a href="#Conclusion" title="Conclusion">Conclusion</a></li>
<li class=""><a href="#References" title="References">References</a></li>
</ul>
</li>
</ul>
</div><div class="toc-menu"><a class="expand-toggle" href="#">Expand all</a><a class="back-to-top" href="#">Back to top</a><a class="go-to-bottom" href="#">Go to bottom</a></div>
            </ul>
        </div>
    </div>
    <div id="ui-toc-affix" class="ui-affix-toc ui-toc-dropdown unselectable hidden-print" data-spy="affix" style="top:17px;display:none;" null null>
        <div class="toc"><ul class="nav">
<li class=""><a href="#BIT-150-Project-2" title="BIT 150: Project 2">BIT 150: Project 2</a><ul class="nav">
<li class=""><a href="#Introduction" title="Introduction">Introduction</a></li>
<li class=""><a href="#Methods" title="Methods">Methods</a><ul class="nav">
<li class=""><a href="#Data-Acquisition" title="Data Acquisition">Data Acquisition</a></li>
<li class=""><a href="#Data-Integration" title="Data Integration">Data Integration</a></li>
<li class=""><a href="#Cytoscape" title="Cytoscape">Cytoscape</a></li>
</ul>
</li>
<li class=""><a href="#Results" title="Results">Results</a><ul class="nav">
<li class=""><a href="#Gene-Diesase-Association-and-Differential-Expression-Table" title="Gene-Diesase Association and Differential Expression Table">Gene-Diesase Association and Differential Expression Table</a></li>
<li class=""><a href="#Cytoscape-Visualization" title="Cytoscape Visualization">Cytoscape Visualization</a></li>
<li><a href="#UCSC-Genome-Browser-Graph" title="UCSC Genome Browser Graph">UCSC Genome Browser Graph</a></li>
<li><a href="#GENT2-Graph-of-Differential-Expression" title="GENT2 Graph of Differential Expression">GENT2 Graph of Differential Expression</a></li>
</ul>
</li>
<li><a href="#Discussion" title="Discussion">Discussion</a></li>
<li><a href="#Conclusion" title="Conclusion">Conclusion</a></li>
<li class=""><a href="#References" title="References">References</a></li>
</ul>
</li>
</ul>
</div><div class="toc-menu"><a class="expand-toggle" href="#">Expand all</a><a class="back-to-top" href="#">Back to top</a><a class="go-to-bottom" href="#">Go to bottom</a></div>
    </div>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.1.1/jquery.min.js" integrity="sha256-hVVnYaiADRTO2PzUGmuLJr8BLUSjGIZsDYGmIJLv2b8=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous" defer></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/gist-embed/2.6.0/gist-embed.min.js" integrity="sha256-KyF2D6xPIJUW5sUDSs93vWyZm+1RzIpKCexxElmxl8g=" crossorigin="anonymous" defer></script>
    <script>
        var markdown = $(".markdown-body");
        //smooth all hash trigger scrolling
        function smoothHashScroll() {
            var hashElements = $("a[href^='#']").toArray();
            for (var i = 0; i < hashElements.length; i++) {
                var element = hashElements[i];
                var $element = $(element);
                var hash = element.hash;
                if (hash) {
                    $element.on('click', function (e) {
                        // store hash
                        var hash = this.hash;
                        if ($(hash).length <= 0) return;
                        // prevent default anchor click behavior
                        e.preventDefault();
                        // animate
                        $('body, html').stop(true, true).animate({
                            scrollTop: $(hash).offset().top
                        }, 100, "linear", function () {
                            // when done, add hash to url
                            // (default click behaviour)
                            window.location.hash = hash;
                        });
                    });
                }
            }
        }

        smoothHashScroll();
        var toc = $('.ui-toc');
        var tocAffix = $('.ui-affix-toc');
        var tocDropdown = $('.ui-toc-dropdown');
        //toc
        tocDropdown.click(function (e) {
            e.stopPropagation();
        });

        var enoughForAffixToc = true;

        function generateScrollspy() {
            $(document.body).scrollspy({
                target: ''
            });
            $(document.body).scrollspy('refresh');
            if (enoughForAffixToc) {
                toc.hide();
                tocAffix.show();
            } else {
                tocAffix.hide();
                toc.show();
            }
            $(document.body).scroll();
        }

        function windowResize() {
            //toc right
            var paddingRight = parseFloat(markdown.css('padding-right'));
            var right = ($(window).width() - (markdown.offset().left + markdown.outerWidth() - paddingRight));
            toc.css('right', right + 'px');
            //affix toc left
            var newbool;
            var rightMargin = (markdown.parent().outerWidth() - markdown.outerWidth()) / 2;
            //for ipad or wider device
            if (rightMargin >= 133) {
                newbool = true;
                var affixLeftMargin = (tocAffix.outerWidth() - tocAffix.width()) / 2;
                var left = markdown.offset().left + markdown.outerWidth() - affixLeftMargin;
                tocAffix.css('left', left + 'px');
            } else {
                newbool = false;
            }
            if (newbool != enoughForAffixToc) {
                enoughForAffixToc = newbool;
                generateScrollspy();
            }
        }
        $(window).resize(function () {
            windowResize();
        });
        $(document).ready(function () {
            windowResize();
            generateScrollspy();
        });

        //remove hash
        function removeHash() {
            window.location.hash = '';
        }

        var backtotop = $('.back-to-top');
        var gotobottom = $('.go-to-bottom');

        backtotop.click(function (e) {
            e.preventDefault();
            e.stopPropagation();
            if (scrollToTop)
                scrollToTop();
            removeHash();
        });
        gotobottom.click(function (e) {
            e.preventDefault();
            e.stopPropagation();
            if (scrollToBottom)
                scrollToBottom();
            removeHash();
        });

        var toggle = $('.expand-toggle');
        var tocExpand = false;

        checkExpandToggle();
        toggle.click(function (e) {
            e.preventDefault();
            e.stopPropagation();
            tocExpand = !tocExpand;
            checkExpandToggle();
        })

        function checkExpandToggle () {
            var toc = $('.ui-toc-dropdown .toc');
            var toggle = $('.expand-toggle');
            if (!tocExpand) {
                toc.removeClass('expand');
                toggle.text('Expand all');
            } else {
                toc.addClass('expand');
                toggle.text('Collapse all');
            }
        }

        function scrollToTop() {
            $('body, html').stop(true, true).animate({
                scrollTop: 0
            }, 100, "linear");
        }

        function scrollToBottom() {
            $('body, html').stop(true, true).animate({
                scrollTop: $(document.body)[0].scrollHeight
            }, 100, "linear");
        }
    </script>
</body>

</html>
